tools/circt-opt/circt-opt.cpp

```diff
@@ -20,58 +26,67 @@
 #include "llvm/Support/InitLLVM.h"
 #include "llvm/Support/ToolOutputFile.h"
 
-using namespace llvm;
 using namespace mlir;
 using namespace circt;
 
-static cl::opt<std::string>
-    inputFilename(cl::Positional, cl::desc("<input file>"), cl::init("-"));
+static llvm::cl::opt<std::string> inputFilename(llvm::cl::Positional,
+                                                llvm::cl::desc("<input file>"),
+                                                llvm::cl::init("-"));
 
-static cl::opt<std::string> outputFilename("o", cl::desc("Output filename"),
-                                           cl::value_desc("filename"),
-                                           cl::init("-"));
+static llvm::cl::opt<std::string>
+    outputFilename("o", llvm::cl::desc("Output filename"),
```

Comment:
Please use clang-format in the LLVM style if you're not already.

Reply:
My setup should be using `{ BasedOnStyle: LLVM, AlwaysBreakTemplateDeclarations: Yes }` (is it the right setup?).  I think most of the reformat that happened here is due to the fact I had to remove the `using namespace llvm;` because of `llvm/IR/Module.h` (included in `mlir/dialect/LLVMIR/LLVMDialect.h`) defining the `Pass` class into the `llvm` namespace, generating a lot of ambiguity errors in the passes registration. Is there maybe a better way to handle those ambiguities other than removing the `using namespace`?

Reply:
I think there's a fix for the Pass namespace issue in MLIR HEAD.

Reply:
Ok, so we should probably update our LLVM submodule to get this patch. I'm not very familiar with submodule management though. What would be the procedure to do this? Start with a PR to the `circt/llvm` repo? What if the update breakes the `circt/circt` master branch?

Reply:
@maerhart I'm working on updating LLVM for you.

Reply:
Ok, thank you for the effort!